tests/test-class-wc-payments-customer-service.php

```diff
@@ -287,4 +287,20 @@ public function test_update_payment_method_with_billing_details_from_order() {
 
 		$this->customer_service->update_payment_method_with_billing_details_from_order( 'pm_mock', $order );
 	}
+
+	public function test_get_payment_methods_for_customer_not_throw_resource_missing_code_exception() {
+		$this->mock_api_client->expects( $this->once() )
+			->method( 'get_payment_methods' )
+			->with( 'cus_test12345' )
+			->willThrowException( new WC_Payments_API_Exception( 'Error Message', 'resource_missing', 400 ) );
+
+		$order = WC_Helper_Order::create_order();
+
+		try {
+			$this->customer_service->get_payment_methods_for_customer( 'cus_test12345', $order );
+			$this->success();
```

Comment:
It could be that your version of the PHPUnit is outdated, but I've updated the code and added an empty check in 11f54e4.

Reply:
Well, the error was also [happening on Travis](https://github.com/Automattic/woocommerce-payments/runs/1101871291), so we would have to fix it regardless of my PHPUnit 😄 

@dwainm - sorry for the nitpick, but `assertEmpty` would also be true if `get_payment_methods_for_customer` returned `null`, `''`, etc. In this case, we need to ensure that the response from `get_payment_methods_for_customer` is the empty array, as its callers iterate on the returned value. Using `assertEmpty` here could mask some bugs since the tests would pass even if that method was modified and returned an invalid value.

Reply:
Ah I ignored the error as it was passing locally so I thought it was related to master.

@luizreis What we're testing is that the error thrown was handled and doesn't bubble up. So if it was an empty array or anything else besides the error it's a pass, that was the reason for simply using the success method.

Reply:
Return values should probably be handle in another test case but I didn't see a reason for testing those with this bug.

Reply:
The other option would be to `assertEquals( true. true )` with a comment, to say that we simply want this test to be marked as passed, as the error was not thrown.

Reply:
Any reason not to test the return value here? It seems to me that testing the value is part of testing the exception not thrown flow, as the fallback is vital to ensure that catching the exception wouldn't throw any other errors up in the chain.

Reply:
> Any reason not to test the return value here?

Maybe I'm being too much of a purist, but the focus should really be on the exception not being thrown. Testing the return value provides no further value in this instance. 

Also, I could have just updated it without adding so many words here, I just wanted to make sure the reason for not testing it was understood and where I wanted to emphasize. Though it doesn't make sense to me I've added it as I trust your judgment on this (a4e086b).

Thanks for your time in reviewing this @luizreis.